### PR TITLE
Feature/merchant one

### PR DESCRIPTION
**Pull request by sagararathi**
Merchant One integration 

Note

In this PR
I have removed ssl version ``` http.ssl_version = :SSLv3``` as this is been removed in original repo here is commit from original repo https://github.com/activemerchant/active_merchant/commit/98188f1b5c74a647ace4260bf2c7866a59e2b909

secondly I have refactored ``` def commit ``` in file  ``` lib/active_merchant/billing/gateways/merchant_one.rb ```  I have removed argument from this method because it's never be used and also it  was adding dependency to other method like ```store, unstore, void ``` which done not require any money to do that api call.

thirdly there are few API calls which does not require any type to be added. so I have added check to ``` def post_data ``` to check if the type is passed ``` nil ``` or empty string to include or exclude type in API call 

Finally I have added bunch of other method's to support void, refund, store, unstore and also added test coverage for all API calls

### CODE REVIEW

File: lib/active_merchant/billing/gateways/merchant_one.rb [hunk 1]
```diff
@@ -42,14 +41,45 @@ def purchase(money, creditcard, options = {})
...
+        post = {}
+        post.merge!(:transactionid => authorization)
+        add_amount(post, money, options)
+        commit('refund', post)
+      end
+
+      def store(creditcard, options = {})
+        post = {}
+        type = nil
+        post['customer_vault'] = 'add_customer'
```
Comment (kilgore5) [inline lib/active_merchant/billing/gateways/merchant_one.rb]: convert this to a function like `add_customer_vault` to follow the patterh of the other functions

Comment (kilgore5) [reply lib/active_merchant/billing/gateways/merchant_one.rb]: also, make consistent with symbols instead of string if possible (across whole file)

**Review by kilgore5 — CHANGES_REQUESTED** [review]

**Review by jGRUBBS — CHANGES_REQUESTED** [review]
@sagararathi can you add the remote tests for the new methods you added to merchant one(void, refund, store, and unstore)? Once you do that this is good to go.

**Comment by sagararathi** [comment]
@jGRUBBS I have added remote test. Please review and let me know if there is any feedback. Thanks @kilgore5 @jGRUBBS

**Comment by jGRUBBS** [comment]
🎉  good work @sagararathi :shipit:

**Pull request merged**